sources/datasources/users.js

```diff
@@ -21,83 +23,75 @@ class UserAPI extends DataSource {
+				address: user.address,
+				mobile: user.mobile,
+				firebaseUID : uid,	
+				emergencyContact: user.emergencyContact,
+				displayPicture: user.displayPicture,
+			});
 		}
-		retPromise = await createdUser.save();
-		return retPromise;
+		return incomingUser;		
 	}
 
-	async updateUser(args) {
-		const userId = args.id;
+	async updateUser(args,uid) {
 		const user = args.user;
 		let retPromise = {};
-		// Create user with basic types;
-		const foundUser = await Users.findById(userId);
-		const originalMemberAccess = foundUser.clubAccess.slice(0);
-
+		const foundUser = await Users.findOne({firebaseUID:uid});
 		let updatedUser = new Users(foundUser);
 		updatedUser = Object.assign(updatedUser, user);
 		updatedUser = new Users(updatedUser);
-		//Add nested types
-		const accessArray = user.clubAccess;
-		if (accessArray != undefined && accessArray.length > 0) {
-			// accessArray exists and is not empty
-			await Promise.all(
-				accessArray.map(async (accessItem, index) => {
-					const clubId = accessItem.club;
-					const foundClub = await Clubs.findById(clubId);
-					const accessObj = {
-						level: accessItem.level,
-						club: foundClub._id,
-						user: userId,
-					};
-					let createdAccessLevel = await AccessLevel.create(accessObj);
-					updatedUser.clubAccess.push(createdAccessLevel._id);
-					foundClub.memberAccess.push(createdAccessLevel._id);
-					await foundClub.save();
-				})
-			);
+		
+		let regex=/^(1|2|3|4|5|7)[0-9][0-9]((AR|AS|BM|BT|CH|CE|CR|CS|CY|EC|EI|EE|ER|FP|HS|ID|LS|MA|ME|MN|MM|PA|PH|SM)|(ar|as|bm|bt|ch|ce|cr|cs|cy|ec|ei|ee|er|fp|hs|id|ls|ma|me|mn|mm|pa|ph|sm))[0-9]{4}$/;
+		if( user.instituteId!=undefined&&foundUser.instituteId==undefined){
+			if(regex.test(user.instituteId)==false){
+				return new ApolloError("Invalid Institute ID");
+			}
+			const email=user.instituteId+'@nitrkl.ac.in';
+			const claimSuccessResponse= await this.updateCustomClaim(email,uid);
+			if(claimSuccessResponse==false){
+				return  new ApolloError("Role Update Failed");;
+			}
 		}
-
 		retPromise = await updatedUser.save();
-		return retPromise;
+		return retPromise;		
 	}
 
-	async deleteUser(id) {
-		const foundUser = await Users.findById(id);
+	async deleteUser(uid) {
+		const foundUser = await Users.find({firebaseUID:uid});;
 		return await foundUser.remove();
 	}
+
+	/**
+	 * Firebase CustomClaim:  Use this function to update users from level 1 to level 2
+	 *  @param {string} email - NITR mail id of the user
+	 */ 
+	async updateCustomClaim(email,uid){
+		const userRecord= await  admin.auth().getUserByEmail(email)		
+		const customClaims={
+			role:"LEVEL2"
+		}
+		admin.auth().setCustomUserClaims(uid, customClaims)
+			.then(() => {
+				console.log("Set "+customClaims.role+" for "+userRecord.displayName+" Success");
+				return true;
+			})
+			.catch(error => {
+				console.log("Set "+customClaims.role+" for "+userRecord.displayName+" Failure");
+				return false;
+			});
+			
+	}
```

Comment:
Hey, who calls this function?
What is the plan? we call from client or Firebase function or what?

Reply:
No we call this function from server. The flow now is we 
check if the update request is for NITR verification
check regex of instituteID
check whether the nitr emailID  exists in Firebase
Upgrade ROLE using Custom Claims(this function)
Update the database
At any point it fails, the nitr verification is rejected

Reply:
So from client's end, we show card to verify, they verify, after they return to app, client calls the endpoint and voila, done?

Am I right?

Reply:
Also, when new custom claim is added, does it over-write old one?
So will user have level1and2 both or just level2?

Reply:
Server calls this function, for some reason I thought it would be better to have the check once more again in backend. It's unnecessary, I've spoke to @Thesmader . 

We could remove it ig

Reply:
@DesignrKnight 
client verifed with firebase and calls the endpoint, voila done!
and it overwrites level1 with level2